expressAPIServer/app.js

```diff
@@ -1,115 +1,92 @@
-      res.status(200).type('application/json').send(rows);
-    }).catch(err => {
-      res.status(500).send("Error querying database");
-    });
+    // Make the SQL query
+    makeSQLQuery(res, query);
 });
 
-app.get('/classes:id', (req, res) => {
+app.get('/getClass', (req, res) => {
+    // Construct the query
     let query = "SELECT * FROM class";
-    let items = []
 
-    let condition = req.query.dept_code
-    if (condition){
-        condArray = condition.split(",");
-        // Add to sql query
-        query = query + " WHERE dept_code = '" + condArray.join("' OR dept_code = '")+ "'";
-    };
+    // Condition where dept_code was provided
+    if (req.query.dept_code){
+        let myArray = req.query.dept_code.split(",");
+        query = query + " WHERE dept_code = '" + myArray.join("' OR dept_code = '")+ "'";
+    }
 
-    // Make the database query
-    new Promise( (resolve, reject) => {
-      con.query(query, (err, result) => {
-          if (err) {
-            reject();
-          } else {
-            resolve(result)
-          }
-      });
-    })
-    .then(rows => {
-      res.status(200).type('application/json').send(rows);
-    }).catch(err => {
-      res.status(500).send("Error querying database");
-    });
+    // Make the SQL query
+    makeSQLQuery(res, query);
 });
 
-app.post('/classes', (req, res) => {
-    let query = "INSERT INTO class (dept_code, class_num, class_name) VALUES ?";
-    data = [[req.body.dept,req.body.num, req.body.name]];
+app.post('/addClass', (req, res) => {
+    // Construct the query
+    let query = "INSERT INTO class (dept_code, class_num, class_name) VALUES (?, ?, ?)";
+
+    // Data from user HTTP request. We pass into makeSQLQuery to prevent injection.
+    data = [req.query.dept_code, req.query.class_num, req.query.class_name];
+
+    makeSQLQuery(res, query, data);
+});
 
-    // Make the database query
+// Make an SQL query
+function makeSQLQuery(res, query, data) {
+    // Promise allows for other items to be completed simulatenously. Check out Javascript
+    // promises for more information.
     new Promise( (resolve, reject) => {
-      con.query(query, (err, result) => {
-          if (err) {
-            reject();
-          } else {
-            resolve(result)
-          }
-      });
+        con.query(query, data, (err, result) => {
+            if (err) {
+                console.log(err)
+                reject();
+            } else {
+                resolve(result)
+            }
+        });
     })
     .then(rows => {
-      res.status(200).type('application/json').send(rows);
+        // Successful, send status 200 and resulting rows
+        res.status(200).type('application/json').send(rows);
     }).catch(err => {
-      res.status(500).send("Error querying database");
+        // Not successful, send status 500, error
+        res.status(500).send("Error querying database");
     });
-
-    res.redirect('/classes:id')
-});
+}
 
 module.exports = app;
```

Comment:
We'll need to compare my API file to yours here- hard to tell what changes are mandatory and which are aesthetic.